miner/miner.go

```diff
@@ -120,8 +120,8 @@ func CloudBase(
 	// Increment the input index of every transaction that has an input in the
 	// new block
 	for _, tx := range b.Transactions[1:] {
```

Comment:
I think this for loop should be removed because a transaction can't reference an input in the same block.